devito/propagator.py

```diff
@@ -197,10 +197,9 @@ def sympy_to_cgen(self, stencils):
         if len(self.factorized) > 0:
             for name, term in zip(self.factorized.keys(), self.factorized):
                 expr = self.factorized[name]
-                # TODO: add support for double precision
-                self.add_local_var(name, "float")
-                # TODO: undo precision enforcing
-                factors.append(cgen.Assign(name, str(ccode(expr.xreplace(self._var_map))).replace("pow", "powf").replace("fabs", "fabsf")))
+                self.add_local_var(name, self.dtype)
+                if self.dtype is np.float32:
+                    factors.append(cgen.Assign(name, str(ccode(expr.xreplace(self._var_map))).replace("pow", "powf").replace("fabs", "fabsf")))
```

Comment:
@navjotk  You said you have a solution to tackle this using proper polymorphism and we agreed to go with this until you're done with the proper fix.  I agree that this is not a very elegant solution and ccode should take care of that internally. 

self.add_local_var(name, self.dtype) creates C vars for the precomputed terms (angles)